apps/sms/views/conversation/js/compose.js

```diff
@@ -51,12 +48,15 @@ var Compose = (function() {
     lastScrollPosition: 0,
     resizing: false,
 
-    // Stop further input because the max size is exceeded
-    locked: false,
-
     // 'sms' or 'mms'
     type: 'sms',
 
+    lock: {
+      send: false,
+      edit: false,
+      type: null
+    },
```

Comment:
> you can pass the accessors also with method injection :)

> When using constructor-based injection we need to have one instance per view; using method-based injection one instance is OK and we just need to inject when we enter the view.

Yeah, I know, I mean that in case of functions/delegates we should pass them only once since ```Compose``` will take care of calling them when it thinks it's needed, so reversing responsibility here, I'm fine with it. So they likely will be injected via ```init``` or similar method, that will be called only once per view.